sys/storage/wifiSettings/wifiSettings.c

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2020 btcven and Locha Mesh developers
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @defgroup     storage_wifiSettings
+ * @ingroup      storage_wifiSettings
+ * @{
+ *
+ * @file        wifiSettings.c
+ * @brief       component created to manage the wifi configuration in memory
+ *
+ * @author      Locha Mesh Developers(conctact@locha.io)
+ * @}
+ */
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+#include "storage/wifiSettings.h"
+#include "storage/nvs.h"
+
+int set_ap_credentials(wifi_settings ap_value){
+
+   int err = nvs_set_string("ap_ssid", ap_value.ssid);
+   if(err != 0){
```

Comment:
can be rephrased as: `if (nvs_set_string("ap_ssid", ap_value.ssid) < 0)`.
p.d.: better use `nvs_set_blob`, otherwise you'll get errors with the null terminator sometimes.